source/sync/migrate-sync-modes.txt

```diff
@@ -0,0 +1,356 @@
+      :alt: Commit a migration
+#. Confirm migration commit by typing "Commit", then select :guilabel:`Commit`.
+   
+   .. image:: /images/confirm-migration-commit.png
+      :alt: Confirm migration commit
+   
+   This will commit your migration and you will not be able to recover your
+   Partition-Based Sync Metadata.
+
+.. _realm-sync-migrate-revert:
+
+Revert Migration
+~~~~~~~~~~~~~~~~
+
+You can revert a migration after it has completed and before it has been committed.
+
+Completed migrations are automatically committed in accordance with your 
+:atlas:`minimum oplog window </atlas/cluster-additional-settings/#set-minimum-oplog-window>`.
+You can increase or decrease the amount of time you have to 
+revert a migration by adjusting the minimum oplog window.
+
+You cannot revert a migration that has been committed.
+
+Similar to canceling a migration, if you revert a migration, you need to 
+start over from the beginning with future migration attempts.
+
+**To revert a migration:** 
+
+#. In the App Services UI, go to your App and select the Device Sync page.
+#. At the top of the page, select the three dots next to 
+   :guilabel:`Manage Sync Migration`.
+#. From the list of options, select 
+   :guilabel:`Revert back to Partition-Based Sync`.
+   
+   .. image:: /images/revert-migration.png
+      :alt: Revert a migration
+   
+   This will revert your migration. You will need to migrate again if you 
+   want to enable Flexible Sync in the future.
+
+.. _realm-sync-migrate-after:
+
+After Backend Migration
+-----------------------
+
+Your migrated backend uses a partition key field in your Realm object models
+to map the Partition-Based Sync client objects to Flexible Sync equivalents on
+the backend. If your object models don't have a partition key field, the
+backend automatically injects one into every object the Partition-Based Sync
+client creates.
+
+These are some other characteristics of the injected field you should be aware
+of:
+
+- The injected field follows this structure: ``<yourPartitionKey> == 
+  <yourPartitionValue>``. Both the field key is derived from your old
+  Partition-Based Sync  **backend**  configuration. The value is derived from
+  your current Partition-Based Sync  **client**  configuration.
+- The injected field does not sync back to the client that created the object.
+  This means the originating client won't be able to read the injected
+  ``partitionKey`` field.
+
+The Partition-Based Sync client automatically creates sync subscriptions for 
+each table where ``partitionKey == partitionValue``. This will continue
+until you :ref:`migrate your client app <realm-sync-migrate-client>` to
+Flexible Sync.
+
+.. _realm-sync-migrate-client:
+
+Migrate Client App to Flexible Sync
+-----------------------------------
+
+After your app's backend has been migrated to Flexible Sync, we recommend
+that you also migrate your client app to Flexible Sync. When you migrate your
```

Comment:
In the SDK side, we should spell out exactly what it means to "migrate your client app to Flexible Sync." I assume that's something like:

- Update a Partition-Based Configuration to a Flexible Sync configuration
- Keep, remove, or update the automatically-generated subscriptions as needed (the Core PR says subscriptions will be given a name in the format "flx_migrated_")
- Add any fields models need to manage subscriptions. i.e. "If your object models don't have a partition key field, the backend automatically injects one into every object the Partition-Based Sync client creates... The injected field does not sync back to the client that created the object." Would we advise devs to specifically add such a key to their models so they're visible and can be updated as needed? And so they remember to give new models the relevant key?

Reply:
Yeah, I think this info will be helpful for the SDK docs. I've captured this info in the Jira ticket for the SDK docs work.